### DISCUSSION TITLE
[.Net] Is there a way to set timeout for container to start ?

### DISCUSSION THREAD

**Discussion by osa0805** [testcontainers/testcontainers-dotnet#1181] [Q&A] [answered]
In TestContainers for Java there is a method when building the container -> "withStartupTimeout". 
I do not see the same method in TestContainers for .Net. 
Is there another way to stops test execution when container is not ready and some timeout was exceeded.
Or is there a plan to implement the same method from Java to .Net ?

**Comment by HofmeisterAn** [comment, marked as answer]
You can simply pass a [CancellationToken](https://learn.microsoft.com/en-us/dotnet/api/system.threading.cancellationtoken) to start.

**Reply by osa0805** [reply]
Great thanks :) I didn't think of that ;<
